pkg/build/api/v1/types.go

```diff
@@ -477,6 +477,13 @@ type SourceBuildStrategy struct {
 
 	// forcePull describes if the builder should pull the images from registry prior to building.
 	ForcePull bool `json:"forcePull,omitempty"`
+
+	// RuntimeImage is the optinal image that is used to run an application
+	// without unneeded dependencies installed.
+	RuntimeImage *kapi.ObjectReference `json:"runtimeImage,omitempty"`
+
+	// RuntimeArtifacts holds the mapping between the artifacts paths on the builder image and the paths on the runtime image where they should be copied.
+	RuntimeArtifacts []ImageSourcePath `json:"runtimeArtifacts,omitempty"`
```

Comment:
> Right now I'm re-using the same Docker instance that we're using for downloading builder image (with the same pull secrets).

yeah i suspect (unfortunately) there's going to be someone who wants a different pull secret for their builder image vs their runtime image and i'd rather make that an option from day one, rather than have the single pull secret represent both values, and then have to make an api change to add a second field later (and some sort of conversion that will copy the single field value into both fields, for legacy users who only specified the one field).

> As far I understand OpenShift doesn't download images in this case, right?

not sure what you mean.  doesn't download what images in what case?